src/libraries/System.Net.WebSockets.Client/tests/WebSocketHelper.cs

```diff
@@ -97,25 +93,48 @@ public static Task<ClientWebSocket> GetConnectedWebSocket(
             Retry(output, async () =>
             {
                 var cws = new ClientWebSocket();
-                configureOptions(cws.Options);
-
-                using (var cts = new CancellationTokenSource(timeOutMilliseconds))
-                {
-                    output.WriteLine("GetConnectedWebSocket: ConnectAsync starting.");
-                    Task taskConnect = invoker == null ? cws.ConnectAsync(server, cts.Token) : cws.ConnectAsync(server, invoker, cts.Token);
-                    Assert.True(
-                        (cws.State == WebSocketState.None) ||
-                        (cws.State == WebSocketState.Connecting) ||
-                        (cws.State == WebSocketState.Open) ||
-                        (cws.State == WebSocketState.Aborted),
-                        "State immediately after ConnectAsync incorrect: " + cws.State);
-                    await taskConnect;
-                    output.WriteLine("GetConnectedWebSocket: ConnectAsync done.");
-                    Assert.Equal(WebSocketState.Open, cws.State);
-                }
+                using var cts = new CancellationTokenSource(timeOutMilliseconds);
+                await ConnectAsync(cws, server, configureOptions, invoker, validateState: true, cts.Token);
                 return cws;
             });
 
+        public static async Task ConnectAsync(
+            ClientWebSocket cws,
+            Uri server,
+            Action<ClientWebSocketOptions> configureOptions,
+            HttpMessageInvoker? invoker = null,
+            bool validateState = true,
+            CancellationToken cancellationToken = default)
```

Comment:
This whole method seems unnecessary, it's used from 2 places which differ by what is happening before and after `ConnectAsync`. Like this is not extracting anything shared apart from calling `ConnectAsync`. Maybe setting `RemoteCertificateValidationCallback`, in which case, my question would be "Why isn't this set up by default".